app/webpacker/components/CompetitionForm/FormSections/DuesEstimate.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Input, Modal } from 'semantic-ui-react';
+import { QueryClient, useQuery } from '@tanstack/react-query';
+import useInputState from '../../../lib/hooks/useInputState';
+import { calculateDuesUrl } from '../../../lib/requests/routes.js.erb';
+import { currenciesData } from '../../../lib/wca-data.js.erb';
+import Loading from '../../Requests/Loading';
+import Errored from '../../Requests/Errored';
+import { fetchJsonOrError } from '../../../lib/requests/fetchWithAuthenticityToken';
+
+const CALCULATE_DUES_QUERY_CLIENT = new QueryClient();
+
+export default function DuesEstimate({
+  close, countryId, currencyCode, baseEntryFee, competitorLimit,
+}) {
+  const [competitorCount, setCompetitorCount] = useInputState(competitorLimit || 0);
+  const currencyInfo = currenciesData.byIso[currencyCode] || currenciesData.byIso.USD;
+
+  const {
+    data, isLoading, isError,
+  } = useQuery({
+    queryKey: ['calculateDues'],
```

Comment:
Does this work? The keys need to include everything that the query depends on. Otherwise, if the query is re-run with a different say base entry fee, then it will retrieve the previous cached data (based on the previous base entry fee), since the keys match. I think.

Reply:
Thanks for that. I've changed it, I'm pretty new to useQuery so didn't knew the theory much.

Apparently this was working with constant queryKey as well apparently. I changed the registration fees and opened the component again, and the output is different, and not from cached. I'm not sure why, maybe in local there is no cache.

Reply:
I think that since the query client is re-created every time the component is first rendered (since it's defined in the dues estimate file rather than wrapping the entire form), so it's a fresh instance with no cache each time? And you can't change any of the keys without first closing the component and then re-opening it fresh.

Generally, I think the best approach is a single query client at the root of the app, but we can't do that since rails is rendering react components individually, so we don't have a single root in the react world (if I understand correctly).

Reply:
That makes sense. I guess these will be changed after the nextJS integration.